File: TypeScript/Practice/LeetCode/Easy/Find the Difference of Two Arrays.ts
// https://leetcode.com/problems/find-the-difference-of-two-arrays/

function findDifference(nums1: number[], nums2: number[]): number[][] {
    const nums1Distinct: number[] = [];
    const nums2Distinct: number[] = [];

    for (let i = 0; i < nums1.length; i++) {
        if (!nums2.includes(nums1[i]) && !nums1Distinct.includes(nums1[i])) {
            nums1Distinct.push(nums1[i]);
        }
    }

    for (let i = 0; i < nums2.length; i++) {
        if (!nums1.includes(nums2[i]) && !nums2Distinct.includes(nums2[i])) {
            nums2Distinct.push(nums2[i]);
        }
    }
    
    return [nums1Distinct, nums2Distinct];    
};

File: C#/Practice/LeetCode/Easy/Find the Difference of Two Arrays.cs
// https://leetcode.com/problems/find-the-difference-of-two-arrays/

public class Solution {
    public IList<IList<int>> FindDifference(int[] nums1, int[] nums2) {
        List<int> nums1Distinct = new List<int>();
        List<int> nums2Distinct = new List<int>();

        for (int i = 0; i < nums1.Length; i++)
        {
            if (!Array.Exists(nums2, num => num == nums1[i]) && !nums1Distinct.Contains(nums1[i]))
            {
                nums1Distinct.Add(nums1[i]);
            }
        }

        for (int i = 0; i < nums2.Length; i++)
        {
            if (!Array.Exists(nums1, num => num == nums2[i]) && !nums2Distinct.Contains(nums2[i]))
            {
                nums2Distinct.Add(nums2[i]);
            }
        }

        return new List<IList<int>> { nums1Distinct, nums2Distinct };
    }
}